test/powershell/engine/Module/TestModuleManifest.ps1

```diff
@@ -24,32 +24,74 @@ Describe "Test-ModuleManifest tests" -tags "CI" {
-            "RequiredModules"="Modules_InvalidRequiredModulesinModuleManifest";
-            "FileList"="Modules_InvalidFilePathinModuleManifest";
-            "ModuleList"="Modules_InvalidModuleListinModuleManifest";
-            "TypesToProcess"="Modules_InvalidManifest";
-            "FormatsToProcess"="Modules_InvalidManifest";
-            #"RootModule"="Modules_InvalidManifest";
-            "ScriptsToProcess"="Modules_InvalidManifest"
+        $args = @{$parameter = "doesnotexist.psm1"}
+        New-ModuleManifest -Path $testModulePath @args
+        Test-Path $testModulePath | Should Be $true
+        [string]$errorId = "$error,Microsoft.PowerShell.Commands.TestModuleManifestCommand"
+
+        { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop } | ShouldBeErrorId $errorId
+    }
+
+    It "module manifest containing valid rootmodule succeeds" -TestCases `
+        @{rootModuleValue = $null},
+        @{rootModuleValue = ""},
+        @{rootModuleValue = "foo.psm1"},
+        @{rootModuleValue = "foo.dll"} {
+
+        param($rootModuleValue)
+
+        New-Item -ItemType Directory -Path testdrive:/module
+        $testModulePath = "testdrive:/module/test.psd1"
+
+        if ($rootModuleValue -ne $null -and $rootModuleValue -ne "")
+        {
+            New-Item -ItemType File -Path testdrive:/module/$rootModuleValue
         }
-        foreach ($parameter in $parametersAndErrors.Keys) {
-            Write-Warning "Testing $parameter"
-            $args = @{$parameter = "doesnotexist.psm1"}
-            New-ModuleManifest -Path $testModulePath @args
-            Test-Path $testModulePath | Should Be $true
-            [string]$errorId = $parametersAndErrors[$parameter] + ",Microsoft.PowerShell.Commands.TestModuleManifestCommand"
-
-            { Test-ModuleManifest -Path $testModulePath -ErrorAction Stop } | ShouldBeErrorId $errorId
-            Remove-Item -Recurse -Force $testModulePath
+        New-ModuleManifest -Path $testModulePath -RootModule $rootModuleValue
+        Test-Path $testModulePath | Should Be $true
+        $moduleManifest = Test-ModuleManifest -Path $testModulePath -ErrorAction Stop
+        $moduleManifest | Should BeOfType System.Management.Automation.PSModuleInfo
+        if ($rootModuleValue -eq $null -or $rootModuleValue -eq "") {
+            $moduleManifest.RootModule | Should BeNullOrEmpty
         }
+        else {
+            $moduleManifest.RootModule | Should Be $rootModuleValue
+        }
+    }
+
+    It "module manifest containing invalid rootmodule returns error" -TestCases `
+        @{rootModuleValue = "foo.psd1"; error = "Modules_InvalidManifest"},
+        @{rootModuleValue = "doesnotexist.psm1"; error = "Modules_InvalidRootModuleInModuleManifest"} {
+
+        param($rootModuleValue, $error)
+
+        New-Item -ItemType Directory -Path testdrive:/module
+        $testModulePath = "testdrive:/module/test.psd1"
+
+        if ($rootModuleValue -ne "doesnotexist.psm1")
```

Comment:
It seems it is better to us to split the tests into two and remove the conditions.